noxfile.py

```diff
@@ -16,21 +16,41 @@ def unit(session):
     session.install("-e", ".[all,dev]", silent=False)
     if PYBOP_SCHEDULED:
         session.run("pip", "install", f"pybamm=={PYBAMM_VERSION}", silent=False)
-    session.run("pytest", "--unit")
+    session.run("pytest", "--unit", "-n", "auto")
 
 
 @nox.session
 def coverage(session):
     session.install("-e", ".[all,dev]", silent=False)
     if PYBOP_SCHEDULED:
         session.run("pip", "install", f"pybamm=={PYBAMM_VERSION}", silent=False)
+    session.run(
+        "pytest", "--unit", "--cov", "--cov-append", "--cov-report=xml", "-n", "auto"
+    )
     session.run(
         "pytest",
-        "--unit",
-        "--examples",
+        "--integration",
         "--cov",
+        "--cov-append",
         "--cov-report=xml",
+        "-n",
+        "auto",
     )
+    session.run("pytest", "--plots", "--cov", "--cov-append", "--cov-report=xml")
+
+
+@nox.session
+def integration(session):
+    session.run_always("pip", "install", "-e", ".[all,dev]")
```

Comment:
I notice that we have been using `session.run_always()`, but this has a chance of not using the correct `pip` (`sys.executable -m pip` would be better) or can even fail in case `pip` is not present on PATH. 

We should try to use `session.install()` throughout `noxfile.py`, because it gives one the option of using a command like `nox -s examples --no-install` which is quite useful for local development, which will skip all the `session.install` commands if provided (but AFAIR it will still run if an existing virtual environment is not present to be reused).